sdk/python/kfp/_client.py

```diff
@@ -200,6 +201,9 @@ def _load_config(self, host, client_id, namespace, other_client_id, other_client
     elif self._is_inverse_proxy_host(host):
       token = get_gcp_access_token()
       self._is_refresh_token = False
+    elif credentials:
+      token = credentials.get_token()
+      config.refresh_api_key_hook = credentials.refresh_api_key_hook
```

Comment:
I think we can move this branch higher, probably only after existing_token. The other branches should be refactored to use the credentials interface afterwards.